src/main/java/net/dries007/tfc/compat/jei/category/SealedBarrelRecipeCategory.java

```diff
@@ -30,29 +41,65 @@ public void setRecipe(IRecipeLayoutBuilder builder, SealedBarrelRecipe recipe, I
         super.setRecipe(builder, recipe, focuses);
         if (recipe.getOnSeal() != null)
         {
-            IRecipeSlotBuilder onSeal = builder.addSlot(RecipeIngredientRole.OUTPUT, 76, 25);
-            onSeal.addIngredients(itemStackProviderIngredient(recipe.getOnSeal(), recipe.getInputItem()));
-        }
-        if (recipe.getOnUnseal() != null)
-        {
-            IRecipeSlotBuilder onUnseal = builder.addSlot(RecipeIngredientRole.OUTPUT, 96, 25);
-            onUnseal.addIngredients(itemStackProviderIngredient(recipe.getOnUnseal(), recipe.getInputItem()));
+            RecipeResult<List<ItemStack>> intermediateItem = itemStackProviderIngredient(recipe.getOnSeal(), recipe.getInputItem());
+            IRecipeSlotBuilder intermediateSlot = builder.addSlot(RecipeIngredientRole.RENDER_ONLY, 76, 5);
+            intermediateSlot.addItemStacks(intermediateItem.result());
+            intermediateSlot.addTooltipCallback((slots, tooltip) -> tooltip.add(1, new TranslatableComponent("tfc.jei.misc.barrel_sealed_full").withStyle(ChatFormatting.DARK_PURPLE, ChatFormatting.ITALIC)));
+            if (!intermediateItem.transforms())
+            {
+                builder.createFocusLink(intermediateSlot, inputItemSlot, outputItemSlot);
+            }
         }
     }
 
     @Override
     public void draw(SealedBarrelRecipe recipe, IRecipeSlotsView recipeSlots, PoseStack stack, double mouseX, double mouseY)
     {
         super.draw(recipe, recipeSlots, stack, mouseX, mouseY);
-
         if (recipe.getOnSeal() != null)
         {
-            slot.draw(stack, 76, 25);
+            slot.draw(stack, 75, 4);
+            arrow.draw(stack, 98, 5);
+            arrowAnimated.draw(stack, 98, 5);
         }
-        if (recipe.getOnUnseal() != null)
+
+        MutableComponent text = getHourText(recipe.getDuration()).withStyle(ChatFormatting.BLACK);
+        Minecraft mc = Minecraft.getInstance();
+        Font font = mc.font;
+        font.draw(stack, text, 74f - font.width(text) / 2.0f, 24f, 0xFFFFFF);
+    }
+
+    @Override
+    protected int arrowPosition(SealedBarrelRecipe recipe)
+    {
+        return recipe.getOnSeal() != null ? super.arrowPosition(recipe) : 63;
+    }
+
+    @Override
+    protected int[] slotPositions(SealedBarrelRecipe recipe)
+    {
+        return recipe.getOnSeal() != null ? new int[] {6, 26, 125, 96} : new int[] {21, 41, 91, 111};
+    }
+
+    private static MutableComponent getHourText(int ticks)
+    {
+        if (ticks < 0)
+        {
+            return new TranslatableComponent("tfc.jei.misc.barrel_sealed");
+        }
+        int hours = (int) Math.ceil((double) ticks / 1000);
+        return new TranslatableComponent("tfc.jei.misc.hours_" + (hours > 1 ? "multiple" : "single"), hours);
```

Comment:
This entire method is redundant and can be replaced with `ICalendar.getTimeDelta()`, which also handles minutes, days, and months.